src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopies = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      stateCopy = addProperties(stateCopy, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      stateCopy = removeProperties(stateCopy, action.keysToRemove);
+    } else if (action.type === 'clear') {
+      stateCopy = clear(stateCopy);
+    }
```

Comment:
Instead of using multiple 'if else' statements, consider using a 'switch' statement for handling different action types as it makes the code more readable and maintainable, especially when you have a limited number of conditions.